src/kbmod/search/trajectory_list.cpp

```diff
@@ -22,6 +22,18 @@ TrajectoryList::TrajectoryList(int max_list_size) {
     gpu_array.resize(max_size);
 }
 
+// Move assignment operator.
+TrajectoryList& TrajectoryList::operator=(TrajectoryList&& other) noexcept {
+    if (this != &other) {
+        max_size = other.max_size;
+        data_on_gpu = other.data_on_gpu;
+        cpu_list = std::move(other.cpu_list);
+        gpu_array = std::move(other.gpu_array);
```

Comment:
I'm not sure this does that correct thing in terms of pointer deallocation. Shouldn't we need to deallocate the current gpu_array?

Reply:
Yep. That would've been a memory leak. Using `set_trajectories()` should completely avoid this